core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -192,19 +197,26 @@ class KafkaRequestHandler(
 
 }
 
+object KafkaRequestHandlerPool {
+  val aggregateThreads = new AtomicInteger(0)
+  val requestHandlerAvgIdleMetricName = "RequestHandlerAvgIdlePercent"
+}
+
 class KafkaRequestHandlerPool(
   val brokerId: Int,
   val requestChannel: RequestChannel,
   val apis: ApiRequestHandler,
   time: Time,
   numThreads: Int,
-  requestHandlerAvgIdleMetricName: String,
   nodeName: String = "broker"
 ) extends Logging {
   private val metricsGroup = new KafkaMetricsGroup(this.getClass)
 
   val threadPoolSize: AtomicInteger = new AtomicInteger(numThreads)
-  /* a meter to track the average free capacity of the request handlers */
+  /* Per-pool idle meter (broker-only or controller-only) */
+  private val perPoolIdleMeterName = nodeName.capitalize + requestHandlerAvgIdleMetricName
```

Comment:
This name is part of Kafka's public API (metric). It would be nice to make this explicit and hard to change by developers.